osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs

```diff
@@ -33,12 +33,18 @@ public class DrawableSpinner : DrawableOsuHitObject
         public SpinnerSpmCounter SpmCounter { get; private set; }
 
         private Container<DrawableSpinnerTick> ticks;
-        private SpinnerBonusDisplay bonusDisplay;
         private PausableSkinnableSound spinningSample;
 
         private Bindable<bool> isSpinning;
         private bool spinnerFrequencyModulate;
 
+        /// <summary>
+        /// The amount of bonus score gained from spinning after the required number of spins, for display purposes.
+        /// </summary>
+        public IBindable<double> GainedBonus => gainedBonus;
+
+        private readonly Bindable<double> gainedBonus = new Bindable<double>();
```

Comment:
not sure i like scoring stuff being here. i think it's actively worse than it was before. explanation for this change?

Reply:
This is done in this way to avoid having to calculate the resultant gained score in both `DefaultSpinner` and `LegacySpinner. 

Both pieces should do exactly nothing but display them with certain styles and animations (same applies to future skinnable spinners).

Reply:
i get that, but i find it messy when jammed into the rest of the spinner like that. i would prefer that `SpinnerBonusDisplay` was brought back and kept as the one thing that's handling this. moreover, i'd also prefer to move [the drawable component, and all of this surrounding layout logic](https://github.com/frenzibyte/osu/blob/779c55d768a6723c1735ef92a26fedaec1dd6760/osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySpinner.cs#L78-L90) in there.

Reply:
That's the one thing that can't be done with inlining legacy coordinates logic to `LegacySpinner`, I don't see how bad it is this is currently in, though. 

Keep in mind, the exact same thing will happen to `SpinnerSpmCounter`, may be even renamed to something like `Calculator`.

It would calculate current SPM privately in `DrawableSpinner`, and the drawable spinner will expose that bindable as read-only (`double` type ofc) for spinner pieces to read from and display them.

Reply:
> That's the one thing that can't be done with inlining legacy coordinates logic to `LegacySpinner`

do you mind explaining why? i do not understand why. it is honestly tiring having to drag this sort of information out. i also do not recall you saying that previously.

the point of composition and splitting out components is to separate concerns. i strongly object to stuffing everything back into `DrawableSpinner`. it is strongly detrimental to code quality.

Reply:
thought it's quite apparent? the legacy bonus counter needs the way `LegacySpinner` is sized and positioned, to place itself perfectly as it was in stable. same applies for legacy spm counter.

moving it out of `LegacySpinner` means having to duplicate the legacy coordinates logic there, and I have [said it](https://github.com/ppy/osu/pull/11975#discussion_r589009286) in the dependency PR that this would happen if going that way.

Reply:
even if the comment you linked was supposed to be a warning that this would happen (which, by the way, i don't read it that way), this would not have happened if the dependency order was correct in the first place.

i've got no interest in continuing this discussion. the last thing i'll say is that `DrawableSpinner` is quite a large class already. if anything, it needs less things in it, and not more.

Reply:
Well, you're saying `DrawableSpinner` needs less things in it, I've moved it into a separate `SpinnerBonusProcessor`, and that's how it looks: https://github.com/ppy/osu/commit/5c7dd3c32e2bc122e88a38ad19d6c8f55b86f8d0

Not sure if there's anything I am able to do any further.